### PR TITLE
Split the amicphys module in preparation for coagulation refactor

### PR DESCRIPTION
**Pull request by huiwanpnnl**
This PR includes the following changes:

- The variables declared at the beginning of `modal_aero_amicphys.F90` are moved to a new module named `modal_aero_amicphys_control.F90`
- Old, pre-ACME/E3SM contents in `modal_aero_coag.F90` are removed.
- Subroutine `mam_coag_1subarea` is moved from `modal_aero_amicphys.F90` to `modal_aero_coag.F90`.

Strictly speaking, these are changes that will not be "felt" by the C++ version of MAM, but they will make the upcoming PR for coagulation refactor easier to review.

### CODE REVIEW

File: components/eam/src/chemistry/modal_aero/aero_model.F90 [hunk 1]
```diff
@@ -2241,7 +2241,7 @@ subroutine aero_model_gasaerexch( loffset, ncol, lchnk, delt, &
 
     use time_manager,          only : get_nstep
     use modal_aero_amicphys,   only : modal_aero_amicphys_intr
-    use modal_aero_coag,       only : modal_aero_coag_sub
+   !use modal_aero_coag,       only : modal_aero_coag_sub
```
Comment (singhbalwinder) [inline components/eam/src/chemistry/modal_aero/aero_model.F90]: Feel free to remove all commented-out code.

File: components/eam/src/chemistry/modal_aero/aero_model.F90 [hunk 2]
```diff
@@ -2395,57 +2395,60 @@ subroutine aero_model_gasaerexch( loffset, ncol, lchnk, delt, &
...
     if (mam_amicphys_optaa <= 0) then
-    ! do gas-aerosol exchange, nucleation, and coagulation using old routines
 
-       ! do gas-aerosol exchange (h2so4, msa, nh3 condensation)
-       if (ndx_h2so4 > 0) then
-          del_h2so4_aeruptk(1:ncol,:) = vmr(1:ncol,:,ndx_h2so4)
-       else
-          del_h2so4_aeruptk(:,:) = 0.0_r8
-       endif
+       call endrun("mam_amicphys_optaa <= 0 is nolonger supported")
```
Comment (singhbalwinder) [inline components/eam/src/chemistry/modal_aero/aero_model.F90]: We should eventually remove the `mam_amicphys_optaa` option as it is >0 for MAM4. We can remove this "if" condition here.

File: components/eam/src/chemistry/modal_aero/aero_model.F90 [hunk 3]
```diff
@@ -2395,57 +2395,60 @@ subroutine aero_model_gasaerexch( loffset, ncol, lchnk, delt, &
...
 
-       call t_startf('modal_gas-aer_exchng')
+  ! ! do gas-aerosol exchange, nucleation, and coagulation using old routines
```
Comment (singhbalwinder) [inline components/eam/src/chemistry/modal_aero/aero_model.F90]: Feel free o remove these commented-out codes.

File: components/eam/src/chemistry/modal_aero/modal_aero_initialize_data.F90 [hunk 4]
```diff
@@ -609,13 +609,14 @@ subroutine modal_aero_initialize(pbuf2d, imozart, species_class)
...
           call modal_aero_amicphys_init( imozart, species_class,n_so4_monolayers_pcage_in )
        else
-          call modal_aero_rename_init
-          !   calcsize call must follow rename call
-          call modal_aero_calcsize_init( pbuf2d, species_class )
-          call modal_aero_gasaerexch_init
-          !   coag call must follow gasaerexch call
-          call modal_aero_coag_init
-          call modal_aero_newnuc_init( mam_amicphys_optaa )
+        ! call modal_aero_rename_init
```
Comment (singhbalwinder) [inline components/eam/src/chemistry/modal_aero/modal_aero_initialize_data.F90]: We don't need this "if" condition anymore.

**Review by singhbalwinder — APPROVED** [review]
Thanks, Hui! I have left some very minor comments. Please run the baseline comparison test suite (`bash mam4_refactor_scripts/compare_baselines_all_tests.sh -b <branch-name> -t <temporary-directory-name>`). If they all PASS, we can merge this and you can start working on the next part.

File: components/eam/src/chemistry/modal_aero/aero_model.F90 (hunk 2 above)
Comment (huiwanpnnl) [reply components/eam/src/chemistry/modal_aero/aero_model.F90]: If block and old codes removed in the next commit.

File: components/eam/src/chemistry/modal_aero/aero_model.F90 (hunk 3 above)
Comment (huiwanpnnl) [reply components/eam/src/chemistry/modal_aero/aero_model.F90]: Done.

File: components/eam/src/chemistry/modal_aero/modal_aero_initialize_data.F90 (hunk 4 above)
Comment (huiwanpnnl) [reply components/eam/src/chemistry/modal_aero/modal_aero_initialize_data.F90]: Removed.

File: components/eam/src/chemistry/modal_aero/aero_model.F90 (hunk 1 above)
Comment (huiwanpnnl) [reply components/eam/src/chemistry/modal_aero/aero_model.F90]: Done.

**Review by singhbalwinder — APPROVED** [review]

**Comment by singhbalwinder** [comment]
Once you run the `compare_baselines_all_tests.sh` tests, please post the results here and we can then merge this PR.

**Comment by huiwanpnnl** [comment]
Hi @singhbalwinder, I have run the `compare_baselines_all_tests.sh` script. Below please find the results:

```
[wanh895@compy01]$ cd /compyfs/wanh895/e3sm_scratch//split_amicphys_ALL_TESTS
[wanh895@compy01]$ ./cs.status.comp_split_amicphys_ALL_TESTS_mam4_org_v2_baselines -s
comp_split_amicphys_ALL_TESTS_mam4_org_v2_baselines: 5 tests
  PASS ERS.ne4pg2_oQU480.F2010.compy_intel
  PASS PEM_P64x1.ne4pg2_oQU480.F2010.compy_intel
  PASS PET.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D_Ln5_P32x1.ne4pg2_oQU480.F2010.compy_intel
  PASS SMS_D.ne4pg2_oQU480.F2010.compy_intel
```

**Comment by singhbalwinder** [comment]
Great! I will merge this PR. I probably need to rebase @mingxuanwupnnl PR on aging but it should be relatively easy to do.

**Pull request merged**